### PR TITLE
Update Netlify extension (add Manage DNS Records)

### PR DESCRIPTION
**Pull request by xmok**
## Description

I have multiple domains in Netlify and I often need to manage the DNS records which is why this PR exists.

For registered domains we can create, delete and view DNS records. I have tried to match the Netlify experience as best as possible so all descriptions, placeholders and labels are copied verbatim from Netlify "Add new record" modal. Since fields can be different I have conditionally validated and toggled visibility of form items. To improve UX, Toasts are shown during and after actions. However, I have not added Toasts to any existing functions.

On a Dev Note,
 - update `api` and `utils`
 - remove `getPreferences` from helpers 
 - remove `Preferences` from interfaces.ts since Preferences is automatically exposed
 - move some variables into a new `constants.ts`

## Screencast

https://github.com/raycast/extensions/assets/30526133/9261a28e-a6d8-4c05-8dc7-955ad7b38ead

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by xmok** [comment]
It seems that Netlify  does not check for the CAA tag to be of the correct type so I have removed that check as it is redundant (plus it would have been better as a Dropdown). I have also moved the new components into a separate `dns` folder to make the components more manageable.

**Review by jasonbarry — APPROVED** [review]
LGTM, thanks for bringing this up to speed @xmok!

File: extensions/netlify/package.json [hunk 1]
```diff
@@ -43,10 +44,10 @@
       "mode": "view"
     },
     {
-      "name": "search-domains",
-      "title": "Search Domains",
+      "name": "search-domains-and-manage-dns-records",
```
Comment (pernielsentikaer) [inline extensions/netlify/package.json:47]: Can we keep the name since hotkeys, aliases and ranking are using the name as ID 🙂

Comment (xmok) [reply extensions/netlify/package.json:47]: Hi.
I'm not really sure what you mean but yeah we can revert the name change if required. Lemme quickly make the change.

Comment (pernielsentikaer) [reply extensions/netlify/package.json]: So the name is the ID for the command, if you change that then the users will lose the ranking / alias / hotkey attached 🙂

Thanks for changing back 🙂

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**